### ISSUE TITLE
Optifine banner cape designer using an outdated banner generator

### ISSUE THREAD

**Issue by CivetKitty** [sp614x/optifine#2476]
So I noticed that Optinine.net still uses the [MinersNeedCoolShoes version](http://www.needcoolshoes.com/banner), but I remember using this site in 1.13 and got a completely different result. I reported this issue in the [MCStacker](https://mcstacker.net/) [discord](https://discord.gg/WCb6GNf) and it now uses the [generator from Minecraft Tools](http://minecraft.tools/en/banner.php). 

This was caused by 2 main issues. The first problem was that the generator from MNCS somehow reversed the color nbt value, making white(0) become black(15), orange(1) become red(14), and so on. Moreover, the banner nbt syntax changed in 1.13 too. For example, instead of `minecraft:banner`, we have `minecraft:white_banner` and such. 

Although MCStacker decided to switch generators, both platform has their downsides. For MNCS, the site uses an outdated syntax. I wanted to contect them to inform this issue, but their twitter was inactive since 2014. Did the admins abandon this site? Can anyone inform this change? As for MCT, the generator follows the survival limit of 6 layers maximum, which is weird considering that they're dealing with commands, not replicating survival stuff. I haven't found a way to contact them either, but at least they followed the change. 

But anyways, Optifine needs to adress this issue and suppport both nbt formats since it supports all versions above 1.7. And since we now have the globe pattern as an option in 1.14, I think it's better to use the newer format.

**Comment by realjackasterisk** [comment]
Honestly, OptiFine should just use its own cape creator instead of relying on other websites that will eventually become outdated, or worse, hacked somehow.

**Comment by realjackasterisk** [comment]
I took a look at the possible options for alternate websites. "Minecraft Tools" banner creator uses the same limitation as vanilla Minecraft banner crafting, so it's not really usable for OptiFine. The only decent option I've found is on Planet Minecraft: https://www.planetminecraft.com/banner/ It has the new colors, the new "globe" pattern, and still works just as well as the MNCS banner creator, albeit with MILDLY slower performance. 

It's also possible to share banner codes, just like MNCS, and the ones on PMC are actually *1* character shorter.

**Comment by Ripstikerpro** [comment]
> Honestly, OptiFine should just use its own cape creator instead of relying on other websites that will eventually become outdated, or worse, hacked somehow.

Yeah, this is probably the best solution. It's just a recipe for disaster to rely on other not associated platforms, with ever decreasing traffic/revenue.
An online editor on Optifine.net would perhaps be a great investment of time.

**Comment by Vexed01** [comment]
> An online editor on Optifine.net would perhaps be a great investment of time.

I believe that someone is working on this (not sure who)

**Comment by Vexed01** [comment]
Ah, it's [ZenithKnight#1225](https://canary.discordapp.com/channels/423430686880301056/426005631997181963/624800994227650562) in [Discord](https://canary.discordapp.com/channels/423430686880301056/426005631997181963/624800994227650562).
However, there is no guarantee this _will_ be added - though I agree it is important.

**Comment by NitroCipher** [comment]
Heyo, ZenithKnight#1225 here. I am, in fact working on a new cape/banner editor for OptiFine.

At this point in time however, it is by no means an official OptiFine project.

I hope, once completed it will be included into the website, but that is a little ways off.

My plan is for it to support the ability to export banner give codes as well. (Compatible with 1.14)

**Comment by cfazilleau** [comment]
Any update on this issue ? It has been there for a long time now ...

**Comment by shardeex** [comment]
Still waiting, he-he.

**Comment by SkiesRed** [comment]
> Heyo, ZenithKnight#1225 here. I am, in fact working on a new cape/banner editor for OptiFine.
> 
> At this point in time however, it is by no means an official OptiFine project.
> 
> I hope, once completed it will be included into the website, but that is a little ways off.
> 
> My plan is for it to support the ability to export banner give codes as well. (Compatible with 1.14)

3 years and waiting, with no update on anything. I only hope something may happen soon

**Comment by smolyoshino** [comment]
yeeuhuh.
:(

**Comment by cfazilleau** [comment]
Optifine is dead anyway...

**Comment by smolyoshino** [comment]
lolwut
"optifine is dead"
>build from november
![image](https://user-images.githubusercontent.com/22896890/213876536-9543ed43-a547-4b45-82d1-1ecaf9450556.png)

**Comment by cfazilleau** [comment]
what I mean is they are not the "only" option for optimized modded minecraft, and given the position of the author towards the open source and modding community, I don't believe it will still be a big player in the domain in the future
